Dockerfile

```diff
@@ -27,15 +27,23 @@ RUN wget https://raw.githubusercontent.com/platformio/platformio-core-installer/
 	source ~/.platformio/penv/bin/activate && \
 	./bin/build-native.sh
 
-FROM frolvlad/alpine-glibc:glibc-2.31
+RUN cp /tmp/firmware/release/meshtasticd_linux_$(uname -m) /tmp/firmware/release/meshtasticd
 
-RUN apk --update add --no-cache g++ shadow && \
-	groupadd -g 1000 mesh && useradd -ml -u 1000 -g 1000 mesh
+##### PRODUCTION BUILD #############
 
-COPY --from=builder /tmp/firmware/release/meshtasticd_linux_x86_64 /home/mesh/
+FROM debian:bookworm-slim
+ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
+
+RUN apt-get update && \
+    apt-get -y install libc-bin libc6 libgpiod2 libyaml-cpp0.7
+RUN groupadd -g 1000 mesh && useradd -ml -u 1000 -g 1000 mesh
 
 USER mesh
 WORKDIR /home/mesh
-CMD sh -cx "./meshtasticd_linux_x86_64 --hwid '${HWID:-$RANDOM}'"
+COPY --from=builder /tmp/firmware/release/meshtasticd /home/mesh/
+
+CMD sh -cx "./meshtasticd -d /home/mesh/data --hwid '${HWID:-$RANDOM}'"
```

Comment:
I wouldn't hard-code a directory here. Instead leave that up to being changed through the configuration file for the ones that need it.

Reply:
This is pretty common practice within a container;  with the intention that a user mounts a docker volume on that path on container start.
 
```
docker run -d \
    -p 4403:4403 \
    --name meshtastic-node \
    -v /opt/meshtastic/data:/home/mesh/data \
    --log-opt mode=non-blocking --log-opt max-buffer-size=4m  \
    meshtastic:v2.3.2
```

The particular path was chosen because the application is ran as the mesh user in the container; so it made sense that the data directory was under the home.

If the volume isn't mounted to that path at start it will be empty;  used and discarded on container remove.